src/pylibsshext/sftp.pyx

```diff
@@ -100,5 +100,10 @@ cdef class SFTP:
                     raise LibsshSFTPException("Opening remote file [%s] for read failed" % remote_file)
         sftp.sftp_close(rf)
 
+    def close(self):
+        if self._libssh_sftp_session is not NULL:
+            sftp.sftp_free(self._libssh_sftp_session)
+            self._libssh_sftp_session = NULL
```

Comment:
I usually like the guard expression style for reducing indents:
```suggestion
        if self._libssh_sftp_session is NULL:
            return
        sftp.sftp_free(self._libssh_sftp_session)
        self._libssh_sftp_session = NULL
```

Reply:
Since this is one level indentation, it shouldn't be an issue. I try to avoid return in between the method whereever possible.

Reply:
Why? `return` does nothing in terms of changing the method's return value.